lib/generators/boring/rswag/install/install_generator.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module Boring
+  module Rswag
+    class InstallGenerator < Rails::Generators::Base
+      DEFAULT_RAILS_PORT = '3000'
+
+      desc "Adds rswag gem to the application"
+
+      class_option :rails_port,
+                   type: :string,
+                   desc: "Tell us the port number where you normally run your rails app. Defaults to PORT #{DEFAULT_RAILS_PORT}",
+                   default: DEFAULT_RAILS_PORT
+
+      def verify_presence_of_rspec_gem
+        gem_file_content_array = File.readlines("Gemfile")
+
+        rspec_is_installed = gem_file_content_array.any? { |line| line.include?("rspec-rails") }
+
+        return if rspec_is_installed
+
+        say "We couldn't find rspec-rails gem which is a dependency for the rswag gem. Please configure rspec and run the generator again!"
+
+        abort
+      end
+
+      def add_rswag_gem
+        say "Adding rswag gem", :green
+
+        run "bundle add rswag"
```

Comment:
Should we default to following gems if rspec test framework in missing?

```
gem 'rswag-api'
gem 'rswag-ui'

group :development, :test do
  gem 'rspec-rails'
  gem 'rswag-specs'
end
```

Reply:
can you try adding this instead in  the generator?

```
def add_rswag_gem
    gem 'rswag'
    gem 'rswag-ui'
    gem 'rswag-specs'
  end

  def install_rswag
    Bundler.with_unbundled_env do
      run 'bundle install'
      generate 'rswag:api:install'
      generate 'rswag:ui:install'
    end
  end
```

Reply:
I am not sure about this, rspec is the dependency for rswag gem so if you don't have rspec installed then this rswag won't work for you. That was the reason why we are checking rspec first with "verify_presence_of_rspec_gem" as well.

Another reason is that your app should always have one test framework configured as well (please correct me if I am wrong on this one.)

Is there a use case I am missing that you are thinking of? Please let me know, in which case I can add the code to run the generator to install rspec as well.

> Should we default to following gems if rspec test framework in missing?
> 
> ```
> gem 'rswag-api'
> gem 'rswag-ui'
> 
> group :development, :test do
>   gem 'rspec-rails'
>   gem 'rswag-specs'
> end
> ```

Reply:
TIL about "gem" method, thanks.

> can you try adding this instead in the generator?
> 
> ```
> def add_rswag_gem
>     gem 'rswag'
>     gem 'rswag-ui'
>     gem 'rswag-specs'
>   end
> 
>   def install_rswag
>     Bundler.with_unbundled_env do
>       run 'bundle install'
>       generate 'rswag:api:install'
>       generate 'rswag:ui:install'
>     end
>   end
> ```